closed/src/java.base/share/native/libjncrypto/NativeCrypto_md.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2019, 2019 All Rights Reserved
+ * ===========================================================================
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * IBM designates this particular file as subject to the "Classpath" exception
+ * as provided by IBM in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, see <http://www.gnu.org/licenses/>.
+ *
+ * ===========================================================================
+ */
+
+#ifndef NATIVECRYPTO_MD_H
+#define NATIVECRYPTO_MD_H
+
+void * load_crypto_library();
```

Comment:
Can we not use `#if` to manage platform-specific code instead of having four nearly identical copies of `NativeCrypto_md.c`?

Reply:
Yes it could be done that way but I think it is cleaner that way it has been done.
I agree the functions are the same / similar and that if we need to modify any then we will have top do it across multiple files but that is something that we will just have to do.